### PR TITLE
Dao #3 Task

### PR DESCRIPTION
**Pull request by IvanLuchkin**
Task completed.

### CODE REVIEW

File: src/main/java/manufacturing/service/ManufacturerServiceImpl.java [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+        return manufacturerDao.create(manufacturer);
+    }
+
+    @Override
+    public Manufacturer get(Long id) {
+        Optional<Manufacturer> manufacturer = manufacturerDao.get(id);
+        if (manufacturer.isEmpty()) {
+            throw new RuntimeException("Manufacturer with id " + id + " not found");
+        }
+        return manufacturer.get();
```
Comment (Vedroid) [inline src/main/java/manufacturing/service/ManufacturerServiceImpl.java]:
```suggestion
        return manufacturerDao.getById(id).orElseThrow(() ->
                        new RuntimeException("Manufacturer with id " + id + " not found"));
```

**Review by Vedroid — APPROVED** [review]

File: src/main/java/manufacturing/service/ManufacturerServiceImpl.java (hunk 1 above)
Comment (IvanLuchkin) [reply src/main/java/manufacturing/service/ManufacturerServiceImpl.java]: Done, thanks!

File: pom.xml [hunk 2]
```diff
@@ -9,7 +9,7 @@
     <version>1.0-SNAPSHOT</version>
```
Comment (Sofasmile) [inline pom.xml:10]: do we need junit dependency in the pom.xml file? let's remove it

File: src/main/java/manufacturing/service/ManufacturerServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+import manufacturing.dao.ManufacturerDao;
+import manufacturing.injections.Inject;
+import manufacturing.injections.Service;
+import manufacturing.model.Manufacturer;
+
+@Service
+public class ManufacturerServiceImpl implements ManufacturerService {
+
+    @Inject
+    ManufacturerDao manufacturerDao;
```
Comment (Sofasmile) [inline src/main/java/manufacturing/service/ManufacturerServiceImpl.java]:
```suggestion
    private ManufacturerDao manufacturerDao;
```

File: src/main/java/manufacturing/dao/ManufacturerDaoImpl.java [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+import manufacturing.db.Storage;
+import manufacturing.injections.Dao;
+import manufacturing.model.Manufacturer;
+
+@Dao
+public class ManufacturerDaoImpl implements ManufacturerDao {
+    public Manufacturer create(Manufacturer manufacturer) {
+        Long id = Storage.manufacturerId++;
+        manufacturer.setId(id);
+        Storage.getManufacturers().put(id, manufacturer);
```
Comment (Sofasmile) [inline src/main/java/manufacturing/dao/ManufacturerDaoImpl.java]: we can move this logic to the Storage class. Example:
```
Storage.saveManufacturer(manufacturer);
```

File: src/main/java/manufacturing/dao/ManufacturerDaoImpl.java [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+        return new ArrayList<>(Storage.getManufacturers().values());
+    }
+
+    public Manufacturer update(Manufacturer newManufacturer) {
+        Manufacturer manufacturerToChange =
+                Storage.getManufacturers().get(newManufacturer.getId());
+        if (manufacturerToChange == null) {
+            throw new RuntimeException("Manufacturer " + newManufacturer.getId() + " not found");
+        }
+        manufacturerToChange.setCountry(newManufacturer.getCountry());
```
Comment (Sofasmile) [inline src/main/java/manufacturing/dao/ManufacturerDaoImpl.java]: better replace newManufacturer in the map:
```
Storage.getManufacturers().put(newManufacturer.getId(), newManufacturer);
```
this code is more flexible. We shouldn't change the logic of this method if the manufacturer object is changed

File: src/main/java/manufacturing/dao/ManufacturerDaoImpl.java [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+        manufacturerToChange.setName(newManufacturer.getName());
+        return manufacturerToChange;
+    }
+
+    public boolean delete(Long id) {
+        Map<Long, Manufacturer> manufacturers = Storage.getManufacturers();
+        Manufacturer manufacturer = manufacturers.get(id);
+        if (manufacturer == null) {
+            throw new RuntimeException("Manufacturer " + id + " not found");
```
Comment (Sofasmile) [inline src/main/java/manufacturing/dao/ManufacturerDaoImpl.java]:
```suggestion
            return false;
```

**Review by Sofasmile — CHANGES_REQUESTED** [review]
good job)

File: src/main/java/manufacturing/dao/ManufacturerDaoImpl.java [hunk 7]
```diff
@@ -10,37 +10,39 @@
...
 
+    @Override
     public Manufacturer update(Manufacturer newManufacturer) {
-        Manufacturer manufacturerToChange =
-                Storage.getManufacturers().get(newManufacturer.getId());
-        if (manufacturerToChange == null) {
-            throw new RuntimeException("Manufacturer " + newManufacturer.getId() + " not found");
+        Map<Long, Manufacturer> manufacturers = Storage.getManufacturers();
+        Long id = newManufacturer.getId();
+        if (!manufacturers.containsKey(id)) {
```
Comment (Sofasmile) [inline src/main/java/manufacturing/dao/ManufacturerDaoImpl.java:33]:
```suggestion
        if (!Storage.getManufacturers().containsKey(id)) {
```

**Review by Sofasmile — APPROVED** [review]

**Pull request merged**